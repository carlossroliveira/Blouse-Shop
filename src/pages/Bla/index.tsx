import React, { useState } from 'react';
import { Link } from 'react-router-dom';

/* import axios from 'axios'; */

export const Bla = () => {
  /* const [users, setUsers] = useState([]);

  useEffect(() => {
    axios
      .get('https://api.github.com/users/carlossroliveira/repos')
      .then((response) => response.data)
      .then((data) => setUsers(data));
  }, []); */

  const [posts, setPosts] = useState([]);

  React.useEffect(() => {
    fetch('http://localhost:3000/api/posts')
      .then((response) => response.json())
      .then((json) => setPosts(json));
  }, []);

  return (
    <>
      <ul>
        {posts.map((post: any) => (
          <>
            <li key={post.id}>
              <Link to={`/posts/${post.id}`}>{post.marca}</Link>
            </li>
          </>
        ))}
      </ul>
    </>
  );
};
