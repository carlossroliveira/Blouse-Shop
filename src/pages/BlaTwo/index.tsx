import React from 'react';
import { useFetch } from '../../hooks/useFetch';
import { useParams } from 'react-router-dom';

export const BlaTwo = () => {
  const { id } = useParams();

  const { data } = useFetch<any>(`http://localhost:5000/posts/${id}`);

  return (
    <>
      <h2 className="cartao__titulo">{data?.marca}</h2>
      <h2 className="cartao__titulo">{data?.descricao}</h2>
    </>
  );
};
