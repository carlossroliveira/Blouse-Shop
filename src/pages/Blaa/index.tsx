import React from 'react';

export const Bla = () => {
  return (
    <>
      <h1>Bla bla</h1>
    </>
  );
};
