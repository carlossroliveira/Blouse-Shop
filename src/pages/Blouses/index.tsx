// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
import { Link } from 'react-router-dom';
// -------------------------------------------------
// Hooks
// -------------------------------------------------
import { useFetch } from '../../hooks/useFetch';

export const Blouses = () => {
  const { data } = useFetch<any>('http://localhost:5000/posts');

  return (
    <>
      <ul>
        {data?.map((post: any) => (
          <>
            <li key={post.id}>
              <Link to={`/posts/${post.id}`}>{post.marca}</Link>
            </li>
          </>
        ))}
      </ul>
    </>
  );
};
