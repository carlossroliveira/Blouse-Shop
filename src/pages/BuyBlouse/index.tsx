// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { useEffect, useState } from 'react';
import { FaShippingFast } from 'react-icons/fa';
import { useNavigate, useParams } from 'react-router-dom';
import { BsArrowLeftCircle, BsArrowRightCircle } from 'react-icons/bs';
// -------------------------------------------------
// Hooks
// -------------------------------------------------
import { useFetch } from '../../hooks/useFetch';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { NamePage } from '../../components/NamePage';
import { ButtonLink } from '../../components/ButtonLink';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { IBProps } from '../../utils/types';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import {
  ContainerSC,
  DivBuySC,
  DivFourSC,
  DivOneSC,
  DivSC,
  DivShieldSC,
  DivSizeSC,
  DivThreeSC,
  DivTwoSC,
  ImgSC,
  ImgShieldSC,
  ParagraphSC,
  SizeButtonSC,
} from './buyBlouseStyles';

export const BuyBlouse = () => {
  const { id } = useParams();
  const navigate = useNavigate();

  const [size, setSize] = useState<string | undefined>('');
  const [count, setCount] = useState<number>(Number(id) + 1);

  const { data } = useFetch<IBProps>(`http://localhost:5000/posts/${id}`);

  const verifyIsSelected = (selected: string) => size === selected;

  const leftFunction = () => count >= 1 && setCount(count - 1);

  const rightFunction = () => count <= 9 && setCount(count + 1);

  useEffect(() => {
    setSize('');
    navigate(`/posts/0${count}`);
  }, [count, navigate]);

  return (
    <ContainerSC>
      <NamePage title="Camisas" />

      <DivSC>
        <DivOneSC>
          <button
            disabled={Number(id) === 1 ? true : false}
            onClick={leftFunction}
          >
            <BsArrowLeftCircle />
          </button>
          <ImgSC
            src={`${data?.foto?.map((item) => item.src)}`}
            alt="Camisa de time"
          />
          <button
            disabled={Number(id) === 9 ? true : false}
            onClick={rightFunction}
          >
            <BsArrowRightCircle />
          </button>
        </DivOneSC>

        <DivTwoSC>
          <DivFourSC>
            {data?.info && (
              <ParagraphSC Info>
                <FaShippingFast /> {data?.info}
              </ParagraphSC>
            )}

            <ParagraphSC Title>
              {data?.foto?.map((item) => item.titulo)}
            </ParagraphSC>

            {data?.preco?.map((item, index) => (
              <>
                <ParagraphSC Inactive key={index}>
                  R$ {item.valorInativo}
                </ParagraphSC>

                <ParagraphSC Installments>R$ {item.valorInicial}</ParagraphSC>

                <ParagraphSC InstallmentsTwo>
                  Ou {item.valorIniParcelado}x de R$ {item.valorInaParcelado}{' '}
                  sem juros
                </ParagraphSC>
              </>
            ))}
          </DivFourSC>

          <DivShieldSC>
            {data?.foto?.map((item) => (
              <ImgShieldSC
                key={item.titulo}
                src={item.shield}
                alt="Escudo do time"
              />
            ))}
          </DivShieldSC>

          <DivThreeSC>
            <ParagraphSC>
              <span>Tamanho: {size}</span>

              <DivSizeSC>
                {data?.tamanho?.slice(0, 1) && (
                  <SizeButtonSC
                    Background={verifyIsSelected(data?.tamanho?.slice(0, 1))}
                    onClick={() => setSize(data?.tamanho?.slice(0, 1))}
                  >
                    {data?.tamanho?.slice(0, 1)}
                  </SizeButtonSC>
                )}

                {data?.tamanho?.slice(3, 4) && (
                  <SizeButtonSC
                    Background={verifyIsSelected(data?.tamanho?.slice(3, 4))}
                    onClick={() => setSize(data?.tamanho?.slice(3, 4))}
                  >
                    {data?.tamanho?.slice(3, 4)}
                  </SizeButtonSC>
                )}

                {data?.tamanho?.slice(6, 7) && (
                  <SizeButtonSC
                    Background={verifyIsSelected(data?.tamanho?.slice(6, 7))}
                    onClick={() => setSize(data?.tamanho?.slice(6, 7))}
                  >
                    {data?.tamanho?.slice(6, 7)}
                  </SizeButtonSC>
                )}

                {data?.tamanho?.slice(9, 11) && (
                  <SizeButtonSC
                    Background={verifyIsSelected(data?.tamanho?.slice(9, 11))}
                    onClick={() => setSize(data?.tamanho?.slice(9, 11))}
                  >
                    {data?.tamanho?.slice(9, 11)}
                  </SizeButtonSC>
                )}

                {data?.tamanho?.slice(13, 16) && (
                  <SizeButtonSC
                    Background={verifyIsSelected(data?.tamanho?.slice(13, 16))}
                    onClick={() => setSize(data?.tamanho?.slice(13, 16))}
                  >
                    {data?.tamanho?.slice(13, 16)}
                  </SizeButtonSC>
                )}

                {data?.tamanho?.slice(18, 22) && (
                  <SizeButtonSC
                    Background={verifyIsSelected(data?.tamanho?.slice(18, 22))}
                    onClick={() => setSize(data?.tamanho?.slice(18, 22))}
                  >
                    {data?.tamanho?.slice(18, 22)}
                  </SizeButtonSC>
                )}
              </DivSizeSC>
            </ParagraphSC>

            <ParagraphSC>
              <span>Tercido:</span> {data?.tecido}
            </ParagraphSC>

            <ParagraphSC>
              <span>Marca:</span> {data?.marca}
            </ParagraphSC>

            <DivBuySC>
              <ParagraphSC>
                <span>Descrição:</span> {data?.descricao}
              </ParagraphSC>

              <ButtonLink
                text="Comprar"
                onClick={() => alert('Comprar')}
                color="black"
              />

              <button onClick={() => alert('Comprar')}>aaaaaaaa</button>
            </DivBuySC>
          </DivThreeSC>
        </DivTwoSC>
      </DivSC>
    </ContainerSC>
  );
};

/* 

position: absolute;
    width: 30rem;
    right: -1rem;
    top: 7rem; 
    
    */
