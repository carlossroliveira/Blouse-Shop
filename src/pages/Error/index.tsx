// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { ButtonLink } from '../../components/ButtonLink';
import { SocialNetworks } from '../../components/SocialNetworks';
// -------------------------------------------------
// Images
// -------------------------------------------------
import photo from './OopsError.svg';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import {
  ContainerSC,
  DivOneSC,
  DivSC,
  DivTwoSC,
  ParagraphSC,
  TitleSC,
} from './errorStyles';

export const Error = (): JSX.Element => {
  return (
    <ContainerSC>
      <DivOneSC>
        <img src={photo} alt="image error" />
      </DivOneSC>

      <DivTwoSC>
        <TitleSC>Esta página não existe!</TitleSC>

        <ParagraphSC>Clique no botão para volta</ParagraphSC>

        <ButtonLink text="Voltar" path="/" />

        <DivSC>
          <SocialNetworks />
        </DivSC>
      </DivTwoSC>
    </ContainerSC>
  );
};
