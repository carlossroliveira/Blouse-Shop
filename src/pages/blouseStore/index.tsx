import React, { useEffect } from 'react';

import axios from 'axios';
import { Mirage } from '../../mirage';

export const BlouseStore = () => {
  /* const [users, setUsers] = useState([]); */

  useEffect(() => {
    axios.get('api/users').then((response) => response.data);
    /* .then((data) => setUsers(data)); */
  }, []);

  Mirage();

  return (
    <>
      <h1>Testing Mirage JS</h1>

      <ul>
        {/* {users?.map((user: any) => (
          <>
            <img
              src={user?.foto?.map((fotos: any) => fotos.src)}
              alt="imagem"
            />
            <div key={user.id}>
              <p>{user?.foto?.map((fotos: any) => fotos.titulo)}</p>

              <p>{user.tecido}</p>
              <p>{user.tamanho}</p>
              <p>{user.preco}</p>
              <p>{user.marca}</p>
              <p>{user.descricao}</p>
            </div>
          </>
        ))} */}
      </ul>
    </>
  );
};
