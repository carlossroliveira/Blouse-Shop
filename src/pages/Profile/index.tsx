// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { FormEvent } from 'react';
import Swal from 'sweetalert2';
import { useNavigate } from 'react-router-dom';
// -------------------------------------------------
// Context
// -------------------------------------------------
import { useAuth } from '../../hooks/useAuth';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { BlouseStore } from '../blouseStore';

export const Profile = () => {
  const auth = useAuth();
  const navigate = useNavigate();

  const handleClose = async (event: FormEvent) => {
    event.preventDefault();

    try {
      auth.logout();
      navigate('/');

      setTimeout(() => {
        const Toast = Swal.mixin({
          toast: true,
          position: 'top-end',
          showConfirmButton: false,
          timer: 3000,
          timerProgressBar: true,
          didOpen: (toast) => {
            toast.addEventListener('mouseenter', Swal.stopTimer);
            toast.addEventListener('mouseleave', Swal.resumeTimer);
          },
        });

        Toast.fire({
          icon: 'success',
          title: 'Deslogado com sucesso!',
        });
      }, 1000);
    } catch (err) {
      err;
    }
  };

  return (
    <>
      <h1>Logado</h1>

      <BlouseStore />

      <button onClick={handleClose}>Sair</button>
    </>
  );
};
