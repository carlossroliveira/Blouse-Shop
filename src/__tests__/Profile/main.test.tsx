// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { Profile } from '../../pages/Profile';
// -------------------------------------------------
// Utils
// -------------------------------------------------
import WithThemeComponent from '../../utils/withThemeComponent';

const LoginWithTheme = WithThemeComponent(Profile);

describe('Testing error profile', () => {
  it('Should render the title correctly', () => {
    render(<LoginWithTheme />);

    expect(
      screen.getByRole('heading', {
        name: /blouse/i,
      }),
    ).toBeInTheDocument();
  });

  it('Should render the buttons correctly', () => {
    render(<LoginWithTheme />);

    expect(
      screen.getByRole('button', {
        name: /lista/i,
      }),
    ).toBeInTheDocument();

    expect(
      screen.getByRole('button', {
        name: /sair/i,
      }),
    ).toBeInTheDocument();
  });

  it('Should render author text correctly', () => {
    render(<LoginWithTheme />);

    expect(screen.getByText(/Â© 2022 \| carlos oliveira/i)).toBeInTheDocument();
  });

  it('component rendering matches snapshot correctly', () => {
    render(<LoginWithTheme />);
    expect(render(<LoginWithTheme />)).toMatchSnapshot();
  });
});
