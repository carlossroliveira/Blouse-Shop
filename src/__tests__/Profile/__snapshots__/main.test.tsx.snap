// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing error profile component rendering matches snapshot correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c2 {
  color: #0000;
  overflow: hidden;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  font-size: 2.5rem;
  line-height: 1.2em;
  text-shadow: 0 0 #383838,0 1.2em #383838;
}

.c2:hover {
  text-shadow: 0 -1.2em #383838,0 0 #383838;
}

.c7 {
  position: absolute;
  right: 8rem;
}

.c8 {
  font-weight: 600;
  font-size: 20px;
  color: #383838;
}

.c0 {
  display: grid;
  grid-template-columns: auto;
  grid-auto-rows: 100px auto 100px;
  grid-template-areas: 'styleOne' 'styleTwo' 'styleThree';
  background-color: #f8f8f8;
}

.c1 {
  grid-area: styleOne;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem 3rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid;
}

.c3 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
}

.c3 ul li + li {
  margin-left: 1rem;
}

.c4 {
  border: none;
  font-size: 1.3rem;
  font-weight: bold;
  -webkit-transition: filter 0.1s;
  transition: filter 0.1s;
  background-color: transparent;
  color: #383838;
}

.c4:hover {
  -webkit-filter: brightness(0.1);
  filter: brightness(0.1);
  cursor: pointer;
}

.c5 {
  grid-area: styleTwo;
  margin-top: 33px;
}

.c6 {
  grid-area: styleThree;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 section {
  position: initial;
}

@media (max-width:975px) {
  .c7 {
    margin-top: 3rem;
    margin-right: -1rem;
  }
}

<body>
    <div>
      <section
        class="c0"
      >
        <header
          class="c1"
        >
          <section
            class=""
          >
            <h1
              class="c2"
            >
              Blouse
            </h1>
          </section>
          <nav
            class="c3"
          >
            <ul>
              <li>
                <button
                  class="c4"
                >
                  Lista
                </button>
              </li>
              <li>
                <button
                  class="c4"
                >
                  Sair
                </button>
              </li>
            </ul>
          </nav>
        </header>
        <main
          class="c5"
        />
        <footer
          class="c6"
        >
          <section
            class="c7"
          >
            <p
              class="c8"
            >
              © 
              2022
               | Carlos Oliveira
            </p>
          </section>
        </footer>
      </section>
    </div>
    <div>
      <section
        class="c0"
      >
        <header
          class="c1"
        >
          <section
            class=""
          >
            <h1
              class="c2"
            >
              Blouse
            </h1>
          </section>
          <nav
            class="c3"
          >
            <ul>
              <li>
                <button
                  class="c4"
                >
                  Lista
                </button>
              </li>
              <li>
                <button
                  class="c4"
                >
                  Sair
                </button>
              </li>
            </ul>
          </nav>
        </header>
        <main
          class="c5"
        />
        <footer
          class="c6"
        >
          <section
            class="c7"
          >
            <p
              class="c8"
            >
              © 
              2022
               | Carlos Oliveira
            </p>
          </section>
        </footer>
      </section>
    </div>
  </body>,
  "container": .c2 {
  color: #0000;
  overflow: hidden;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  font-size: 2.5rem;
  line-height: 1.2em;
  text-shadow: 0 0 #383838,0 1.2em #383838;
}

.c2:hover {
  text-shadow: 0 -1.2em #383838,0 0 #383838;
}

.c7 {
  position: absolute;
  right: 8rem;
}

.c8 {
  font-weight: 600;
  font-size: 20px;
  color: #383838;
}

.c0 {
  display: grid;
  grid-template-columns: auto;
  grid-auto-rows: 100px auto 100px;
  grid-template-areas: 'styleOne' 'styleTwo' 'styleThree';
  background-color: #f8f8f8;
}

.c1 {
  grid-area: styleOne;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem 3rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid;
}

.c3 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
}

.c3 ul li + li {
  margin-left: 1rem;
}

.c4 {
  border: none;
  font-size: 1.3rem;
  font-weight: bold;
  -webkit-transition: filter 0.1s;
  transition: filter 0.1s;
  background-color: transparent;
  color: #383838;
}

.c4:hover {
  -webkit-filter: brightness(0.1);
  filter: brightness(0.1);
  cursor: pointer;
}

.c5 {
  grid-area: styleTwo;
  margin-top: 33px;
}

.c6 {
  grid-area: styleThree;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 section {
  position: initial;
}

@media (max-width:975px) {
  .c7 {
    margin-top: 3rem;
    margin-right: -1rem;
  }
}

<div>
    <section
      class="c0"
    >
      <header
        class="c1"
      >
        <section
          class=""
        >
          <h1
            class="c2"
          >
            Blouse
          </h1>
        </section>
        <nav
          class="c3"
        >
          <ul>
            <li>
              <button
                class="c4"
              >
                Lista
              </button>
            </li>
            <li>
              <button
                class="c4"
              >
                Sair
              </button>
            </li>
          </ul>
        </nav>
      </header>
      <main
        class="c5"
      />
      <footer
        class="c6"
      >
        <section
          class="c7"
        >
          <p
            class="c8"
          >
            © 
            2022
             | Carlos Oliveira
          </p>
        </section>
      </footer>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
